### PR TITLE
feat: 🎸 Setup types in cred library forms and use HDS component

### PR DESCRIPTION
**Pull request by ZedLi**
:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-7673)

## Description
Initial change to allow credential libraries to have types. Also converted normal `vault` 
### Screenshots (if appropriate):

image

image

### CODE REVIEW

File: ui/admin/app/routes/scopes/scope/credential-stores/credential-store/credential-libraries/new.js [hunk 1]
```diff
@@ -13,12 +16,22 @@ export default class ScopesScopeCredentialStoresCredentialStoreCredentialLibrari
...
    */
   model() {
-    const { id: credential_store_id, type } = this.modelFor(
+    const { id: credential_store_id } = this.modelFor(
       'scopes.scope.credential-stores.credential-store'
     );
+
     return this.store.createRecord('credential-library', {
-      type,
+      type: TYPE_CREDENTIAL_LIBRARY_VAULT_GENERIC,
```
Comment (randallmorey) [inline ui/admin/app/routes/scopes/scope/credential-stores/credential-store/credential-libraries/new.js:24]: We still need to get the user-selected type from the query param, don't we?  Generic is the default if type is not provided.

**Review by randallmorey — CHANGES_REQUESTED** [review]

File: ui/admin/app/routes/scopes/scope/credential-stores/credential-store/credential-libraries/new.js (hunk 1 above)
Comment (ZedLi) [reply ui/admin/app/routes/scopes/scope/credential-stores/credential-store/credential-libraries/new.js]: I was going to add in that functionality in the next PR but I can add it here

**Review by calcaide — APPROVED** [review]
LGTM!! Awesome work!!

File: addons/core/translations/resources/en-us.yaml [hunk 2]
```diff
@@ -459,6 +459,9 @@ credential-library:
   types:
     vault-generic: Generic Secrets
     vault-ssh-cert: SSH Certificate Credentials
+  help:
+    vault-generic: Authorize sessions using credentials brokered by Vault
```
Comment (randallmorey) [inline addons/core/translations/resources/en-us.yaml:463]: Can we describe about the type without talking about authorization or sessions?

**Review by randallmorey — CHANGES_REQUESTED** [review]

File: addons/core/translations/resources/en-us.yaml (hunk 2 above)
Comment (ZedLi) [reply addons/core/translations/resources/en-us.yaml:463]: Do you have any suggestions? I'm not sure what else it would say

**Review by DhariniJeeva — APPROVED** [review]

File: addons/core/translations/resources/en-us.yaml (hunk 2 above)
Comment (ZedLi) [reply addons/core/translations/resources/en-us.yaml:463]: I'm going to defer any rewording for later once I can get some more input from others as well and merge it for now

**Pull request merged**